gh-137384: fix warnings finalization order (#138027)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-08-22-11-39-40.gh-issue-137384.j4b_in.rst

> Fix a crash when :mod:`warnings` state was finalised too early. Patch by Kumar Aditya.

Reviewer: Up to you, but I think it's generally better to say *what* was fixed rather than *how*, because it's more useful to users reading the changelog:
```suggestion
Fix a crash when using the :mod:`warnings` module in a finalizer at shutdown. Patch by Kumar Aditya.
```
